server/src/services/plugin-execution.service.ts

```diff
@@ -0,0 +1,214 @@
+import { Plugin as ExtismPlugin, newPlugin } from '@extism/extism';
+import { Injectable } from '@nestjs/common';
+import { resolve } from 'node:path';
+import { PLUGIN_JWT_SECRET } from 'src/constants';
+import { Asset, WorkflowAction, WorkflowFilter } from 'src/database';
+import { OnEvent, OnJob } from 'src/decorators';
+import { JobName, JobStatus, QueueName } from 'src/enum';
+import { ArgOf } from 'src/repositories/event.repository';
+import { PluginContext, pluginTriggers, PluginTriggerType } from 'src/schema/tables/plugin.table';
+import { BaseService } from 'src/services/base.service';
+import { PluginHostFunctions } from 'src/services/plugin-host.functions';
+import { IWorkflowJob, JobItem, JobOf, WorkflowData } from 'src/types';
+import { TriggerConfig } from 'src/types/plugin-schema.types';
+
+interface WorkflowContext {
+  jwtToken: string;
+  asset: Asset;
+  triggerConfig: TriggerConfig | null;
+}
+
+@Injectable()
+export class PluginExecutionService extends BaseService {
+  private loadedPlugins: Map<string, ExtismPlugin> = new Map();
+  private hostFunctions!: PluginHostFunctions;
+
+  @OnEvent({ name: 'AppBootstrap' })
+  async onBootstrap() {
+    this.hostFunctions = new PluginHostFunctions(
+      this.assetRepository,
+      this.albumRepository,
+      this.accessRepository,
+      this.cryptoRepository,
+      this.logger,
+    );
+    await this.loadCorePlugins();
+  }
+
+  private async loadCorePlugins() {
+    const plugins = await this.pluginRepository.getAllPlugins();
+    for (const plugin of plugins) {
+      try {
+        const pluginPath = resolve(process.cwd(), '..', plugin.wasmPath);
+        this.logger.debug(`Loading plugin: ${plugin.name} from ${pluginPath}`);
+
+        const extismPlugin = await newPlugin(pluginPath, {
+          useWasi: true,
+          functions: this.hostFunctions.getHostFunctions(),
+        });
+
+        this.loadedPlugins.set(plugin.id, extismPlugin);
+        this.logger.log(`Successfully loaded plugin: ${plugin.name}`);
+      } catch (error) {
+        this.logger.error(`Failed to load plugin ${plugin.name}:`, error);
+      }
+    }
+  }
+
+  @OnEvent({ name: 'AssetCreate' })
+  async handleAssetCreate({ asset }: ArgOf<'AssetCreate'>) {
+    const trigger = pluginTriggers.find((trigger) => trigger.type === PluginTriggerType.AssetCreate);
```

Comment:
this whole method is actually going to be the same for every trigger.
- Resolve trigger
- Resolve `ownerId` from event
- Find workflows for trigger + `ownerId`
- Create workflow event/data payload from incoming event
- Queue workflow run jobs

Can you extract this into something that looks like this?
```ts
async handleAssetCreate({ asset }: ArgOf<'AssetCreate'>) {
  await this.handleTrigger(PluginTriggerType.AssetCreate, { ownerId: asset.ownerId, event: { assetId: asset.id } });
}